packages/core/src/external/mpi/patient-blocker.ts

```diff
@@ -0,0 +1,20 @@
+import { GenderAtBirth, PatientDataMPI } from "./patient";
+
+export type PatientBlock = {
+  cxId?: string;
+  facilityIds?: string[];
+  data?: {
+    dob?: string;
+    genderAtBirth?: GenderAtBirth;
+    firstNameInitial?: string;
+    lastNameInitial?: string;
+  };
+};
+
+export abstract class PatientBlocker {
+  abstract block(data: PatientBlock): Promise<PatientDataMPI[]>;
+}
+
+export function makeBlockerFactory(blockerType: new () => PatientBlocker): PatientBlocker {
+  return new blockerType();
+}
```

Comment:
factory approach

Reply:
A factory is a function that abstracts away the implementation that's being used, usually based on some environment parameter/setting.

In this case, we're passing the implementation, so the factory is just a fancy "instantiator" - no need for it.

Reply:
Great. I knew I was forcing something with the factory bit at the end there, but happy the interface part is enough :)